src/lib/warn-upload-overwrite.js

```diff
@@ -15,139 +16,195 @@ const responseChoicesWithoutDiff = [
 ];
 const responseChoicesWithDiff = responseChoicesWithoutDiff.concat([ 'View diff' ]);
 
-const getRevsAndHashesDocKey = () => {
+const getEnvironmentKey = () => {
   const parsed = url.parse(environment.apiUrl);
   const key = `${parsed.hostname}${parsed.pathname || 'medic'}`;
   return key;
 };
 
-const preUploadByRev = async (db, doc) => {
-  let localDoc = JSON.parse(JSON.stringify(doc));
+const getBookmarksDir = () => path.join(environment.pathToProject, '.bookmarks');
 
-  // Pull remote _rev
-  let remoteRev;
-  let remoteDoc;
-  try {
-    remoteDoc = await db.get(localDoc._id);
-    remoteRev = remoteDoc._rev;
-  } catch (e) {
-    // continue regardless of error
-    log.trace('Trying to fetch remote _rev', e);
+const getStoredHashes = () => {
+  const bookmarksDir = getBookmarksDir();
+  if (!fs.exists(bookmarksDir)) {
+    fs.mkdir(bookmarksDir);
   }
 
-  // Pull local _rev
-  let localRev;
-  try {
-    localRev = JSON.parse(fs.read(`${environment.pathToProject}/._revs/${localDoc._id}.json`).trim())[getRevsAndHashesDocKey()];
-  } catch (e) {
-    // continue regardless of error
-    log.trace('Trying to fetch local _rev', e);
+  const filePath = path.join(bookmarksDir, 'hashes.json');
+  if (fs.exists(filePath)) {
+    try {
+      return JSON.parse(fs.read(filePath).trim());
+    } catch(e) {
+      log.info('Error trying to read bookmark, continuing anyway', e);
+    }
   }
```

Comment:
This change stores all data in a single json file to get around having to support all the possible filesystem inconsistencies.

Reply:
Please don't let backward compatibility stop you from making this feature something that works for TLs. One file is a huge improvement for our pull request clarity and for all the merge conflicts this feature is going to cause. 

Most people on my team gitignored the hash files or regard the feature is completely broken. LG/D-Tree use medic-conf 2.x.

Reply:
Perhaps hashes for `localhost/medic` could go into its own file and that file could be safely gitignored?

Reply:
I like that - I'll split the local and remote hashes so one can be safely gitignored. I've also decided to format the hashes file so that it's easier to read and conflicts are easier to resolve.

Reply:
Done.